features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/factories/event/TimelineItemEventFactory.kt

```diff
@@ -71,6 +71,14 @@ class TimelineItemEventFactory @Inject constructor(
             url = senderAvatarUrl,
             size = AvatarSize.TimelineSender
         )
+        // Send state based on local data
+        fun localSendState(): EventSendState {
+            return currentTimelineItem.event.localSendState ?: EventSendState.NotSentYet
+        }
+        // Send state: if the event has an event id, then it's already sent
+        val processedSendState = currentTimelineItem.eventId?.let {
+            currentTimelineItem.event.localSendState ?: EventSendState.Sent(it)
+        } ?: localSendState()
```

Comment:
I think 'send state' in the SDK is only applicable to local items, so wondering if it could be a bit confusing to fill a send state for a non-local timeline item on the Android side?

Perhaps something more like `isRemote` or `canReact`?

Also, FYI, the logic might go out of date after https://github.com/matrix-org/matrix-rust-sdk/issues/2127 so perhaps worth centralising it in the Rust SDK

Reply:
`isRemote` sounds good, it's the same thing I wanted to achieve with `sendState` vs `localSendState` here, but it's easier to understand. My only problem with it is we're defaulting `localSendState` to `NotSent` here, so you can have a remote event with 'not sent' local send state, which is a bit contradictory. Maybe I can make it nullable and not default to anything, or make it a sealed interface `SendState { Local(...), Remote(eventId) }`.

> Also, FYI, the logic might go out of date after https://github.com/matrix-org/matrix-rust-sdk/issues/2127 so perhaps worth centralising it in the Rust SDK

I find it a bit weird wanting to apply reactions to local echoes, but good to know. Thanks for the heads up!

Reply:
> Maybe I can make it nullable and not default to anything

That sounds like a good idea; this keeps it better aligned with the SDK and there are some events where 'send state' does not apply (i.e. events not sent by the user)

Reply:
> I find it a bit weird wanting to apply reactions to local echoes

I think the use case here is for sending a message pre-populated with emoji reactions as a prompt e.g:

```
Should we order pizza tonight?
👍 👎 🤷
```